lib/logstash/outputs/http.rb

```diff
@@ -182,16 +156,21 @@ def send_events(events)
     successes = java.util.concurrent.atomic.AtomicInteger.new(0)
     failures  = java.util.concurrent.atomic.AtomicInteger.new(0)
     retries = java.util.concurrent.atomic.AtomicInteger.new(0)
-    
+    event_count = @format == "json_batch" ? 1 : events.size
+
     pending = Queue.new
-    events.each {|e| pending << [e, 0]}
+    if @format == "json_batch"
+      pending << [events, 0, true]
```

Comment:
By setting a `@is_batch` variable (w/ a `is_batch` accessor) to true or false based on the `:format` parameter during the `register` method we can avoid passing it around in multiple places, greatly simplifying the code. wdyt?